test/mlir/onnx/onnx_lowering.mlir

```diff
@@ -1998,6 +1998,47 @@ func private @test_clip_default_min(%arg0: tensor<3xf32>, %arg1: tensor<f32>, %a
 
 // -----
 
+// Test whether the lowering is correct in the presence of dynamic dimensions.
```

Comment:
If you use ONNXReshape and ONNXTranspose, you may find this helpful: https://github.com/onnx/onnx-mlir/blob/master/src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp#L126 for DepthToSpace. It stops the lowering at a middle state, then the lit test output would be simpler, containing ONNXReshape and ONNXTranspose.